spec/ControllerSpec.js

```diff
@@ -0,0 +1,116 @@
+(function() {
+  console.log("Testing Controller: ");
+
+  var mockPromoListModel = {
+    addArticleCalledCount: 0,
+    articles: [],
+    addArticle: function(article) {
+      this.articles.push(article);
+      this.addArticleCalledCount++;
+    }
+  };
+
+  var MockPromoListModel = function() {
+    return mockPromoListModel;
+  };
+
+  var mockPromoListView = {
+    toHtmlCalledCount: 0,
+    toHtml: function() {
+      this.toHtmlCalledCount++;
+      return "<div>Litany of fascinating article headlines</div>";
+    }
+  };
+
+  var MockPromoListView = function(promoListModel) {
+    mockPromoListView.promoListModel = promoListModel;
+    return mockPromoListView;
+  };
+
+  var mockRequestInstance = {
+    openCalledCount: 0,
+    sendCalledCount: 0,
+    open: function(method, location, asynchronousBoolean) {
+      this.methodCalled = method;
+      this.location = location;
+      this.asynchronousBoolean = asynchronousBoolean;
+      this.openCalledCount++;
+    },
+    send: function() {
+      this.sendCalledCount++;
+    },
+    readyState: 'yep',
+    status: 200,
+    responseText: sampleJson
+  };
+
+  var MockRequestClass = function() {
+    return mockRequestInstance;
+  };
+
+  MockRequestClass.DONE = 'yep';
+
+  var controller = new Controller(MockPromoListModel, MockPromoListView, MockRequestClass);
+
+  (function() {
+    console.log(" it delivers an instance of PromoListModel to PromoListView on instantiation");
+    assert.isTrue(mockPromoListView.promoListModel === mockPromoListModel);
```

Comment:
Nitpicking, and not sure if this is within the bounds of the test framework you wrote, but generally assert.areEqual communicates the intent better when comparing two things.